software/controller/lib/debug/eeprom_cmd.cpp

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2020-2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "commands.h"
+
+namespace Debug::Command {
+
+ErrorCode EepromHandler::Process(Context *context) {
+
+  // We have at least subcommand and address (3 bytes)
+  if (context->request_length < 3)
+    return ErrorCode::kMissingData;
+
+  // Whatever the subcommand, bytes 1 and 2 are the address
+  uint16_t address = u8_to_u16(&context->request[1]);
+
+  // Process subcommand
+  if (context->request[0] == 0) {
+    // Read command requires length (bytes 3 and 4)
+    if (context->request_length < 5)
+      return ErrorCode::kMissingData;
+    uint16_t length = u8_to_u16(&context->request[3]);
+    if (length > context->max_response_length)
+      return ErrorCode::kNoMemory;
+
+    if (eeprom_->ReadBytes(address, length, context->response,
+                           context->processed)) {
+      // only set response_length if the read has been successfully sent
+      context->response_length = length;
+    }
+    return ErrorCode::kWait;
+
+  } else if (context->request[0] == 1) {
+    // at least one byte of data is given after the address
+    if (context->request_length < 4)
+      return ErrorCode::kMissingData;
+
+    // length of data to be written is the length of the request minus
+    // the subcommand and address bytes
+    uint16_t length = static_cast<uint16_t>(context->request_length - 3);
+
+    if (length > kMaxWriteLength)
```

Comment:
Tracking down where `kMaxWriteLength` is defined made me realize the definitions for these handler subclasses are all in one header, and that two of them are even implemented there. Any chance we could move those implementations out into cpps and thus get them tested as well? Also, not sure if I am just being dogmatic about this, but should the definitions all be in one header?